packages/contracts-bedrock/src/L1/L1CrossDomainMessenger.sol

```diff
@@ -31,23 +35,44 @@ contract L1CrossDomainMessenger is CrossDomainMessenger, ISemver {
     address private spacer_253_0_20;
 
     /// @notice Semantic version.
-    /// @custom:semver 2.6.0
-    string public constant version = "2.6.0";
+    /// @custom:semver 2.7.0
+    string public constant version = "2.7.0";
+
+    /// @notice Contract of the SystemConfig.
+    ISystemConfig public systemConfig;
 
     /// @notice Constructs the L1CrossDomainMessenger contract.
-    constructor() {
+    constructor() ReinitializableBase(2) {
         _disableInitializers();
     }
 
     /// @notice Initializes the contract.
-    /// @param _superchainConfig Contract of the SuperchainConfig contract on this network.
+    /// @param _systemConfig Contract of the SystemConfig contract on this network.
     /// @param _portal Contract of the OptimismPortal contract on this network.
-    function initialize(ISuperchainConfig _superchainConfig, IOptimismPortal _portal) external initializer {
-        superchainConfig = _superchainConfig;
+    function initialize(ISystemConfig _systemConfig, IOptimismPortal _portal) external reinitializer(initVersion()) {
+        systemConfig = _systemConfig;
         portal = _portal;
         __CrossDomainMessenger_init({ _otherMessenger: CrossDomainMessenger(Predeploys.L2_CROSS_DOMAIN_MESSENGER) });
     }
 
+    /// @notice Upgrades the contract to have a reference to the SystemConfig.
+    /// @param _systemConfig The new SystemConfig contract.
+    function upgrade(ISystemConfig _systemConfig) external reinitializer(initVersion()) {
+        systemConfig = _systemConfig;
```

Comment:
small nit but I'd perfer to zero the spacer before setting the system config, lets us use the gas from the refund to write the systemconfig